src/holoscan_cli/utils/host_setup.py

```
@@ -205,26 +205,51 @@ def get_ubuntu_codename() -> str:
     )
+    if dry_run:
+        info(f"[dryrun] Would fetch the Kitware archive key -> {keyring_path}")
+        write_system_file("/etc/apt/sources.list.d/kitware.list", source_line, dry_run=True)
+    else:
+        # Fetch + dearmor the key as the invoking user; install the keyring as
+        # root. Two separate steps (not a shell pipeline) so a download failure
+        # cannot be masked by the pipeline's exit status and produce an empty
+        # keyring.
+        try:
+            key = subprocess.run(
+                ["wget", "-qO-", "https://apt.kitware.com/keys/kitware-archive-latest.asc"],
```

Ruff (0.15.20)

[error] Starting a process with a partial executable path

(S607)